### PR TITLE
Fix cross section bug

### LINKED ISSUE #943: Higlass breaks after adding a horizontal or vertical rule
_Commit_: https://higlass.io

#### Steps to reproduce

Jun-29-2020 18-40-53

#### Observed behavior
After adding the rule the following error pops up
```
TypeError: Cannot read property 'getMinNonZeroInSubset' of undefined
```
When you then click on a track's cog wheel icon the following error pops up and the page goes blank:
```
TypeError: Cannot read property 'domain' of undefined
```

#### Expected behavior
- Should not happen

### PR DESCRIPTION
**Pull request by alexander-veit**
## Description

> What was changed in this pull request?

Added `denseDataExtrema` to cross section data fetcher.

> Why is it necessary?

Fixes #943 

## Checklist

- [x] Set proper GitHub labels (e.g. v1.6+, ignore if you don't know)
- [x] Unit tests added or updated
- [ ] Documentation added or updated
- [ ] Example(s) added or updated
- [ ] Update schema.json if there are changes to the viewconf JSON structure format
- [ ] Screenshot for visual changes (e.g. new tracks or UI changes)
- [x] Updated CHANGELOG.md

### CODE REVIEW

**Comment by alexander-veit** [comment]
Tests are failing, but that does not seem to be related to this PR. Tests are also failing on `develop`.

**Comment by pkerpedjiev** [comment]
Thanks, I think those test failures are because of my PR. I'll try to fix them later today.

File: app/scripts/DataFetcher.js [hunk 1]
```diff
@@ -447,6 +447,7 @@ export default class DataFetcher {
         const newTile = {
           min_value: Math.min.apply(null, dataSlice),
           max_value: Math.max.apply(null, dataSlice),
+          denseDataExtrema: new DenseDataExtrema1D(dataSlice),
```
[the remark this answers is not in the document]
Comment (alexander-veit) [reply app/scripts/DataFetcher.js:450]: Yes, that's the fix. Cross sections take a slice of the 2D matrix and transform that into a bar chart. The bar chart needs to know how to compute minimas and maximas of arbitrary subsets. That's what `DenseDataExtrema1D` is providing.
Thus, cross sections do depend on numerical data - or am I misunderstanding something?

[the remark this answers is not in the document]
Comment (flekschas) [reply app/scripts/DataFetcher.js:450]: No worries. I thought those are just point annotations drawing a dashed line at a specific location. I didn't know they do more than that. Now I am curious how I can do `transform that into a bar chart` through the UI :) Anyway, this is unrelated.

[the remark this answers is not in the document]
Comment (alexander-veit) [reply app/scripts/DataFetcher.js:450]: What you mean are "Horizontal/Vertical rules". These should be indeed independent of the data.
When you add a cross section via the context menu, a bar chart with a custom data fetcher is added (probably by modifying the viewconf). The data fetcher points to the row (or column) of the matrix that you want to display.

**Pull request merged**